### PR TITLE
Stop tailing files after no updates to them for X minutes

### LINKED ISSUE #1761: Stop tailing files after no updates to them for X minutes
It would be great if there was a setting so that promtail would stop tailing files after they not received any entries for X amount of time.    I don't see anything like that in here.  Our log collection server has literally 100's of thousands of files and as it sits promtail crashes if it tries to tail too many files.

### PR DESCRIPTION
**Pull request by adityacs**
**What this PR does / why we need it**:
Stop tailing files after no updates to them for X minutes

**Which issue(s) this PR fixes**:
Fixes #1761 

**Special notes for your reviewer**:
Just first version. Needs discussion.

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Comment by adityacs** [comment]
@slim-bean What's your take on this? Should we go ahead with this feature?

File: pkg/promtail/targets/filetarget.go [hunk 1]
```diff
@@ -304,7 +306,16 @@ func (t *FileTarget) stopTailing(ps []string) {
 func toStopTailing(nt []string, et map[string]*tailer) []string {
 	// Make a set of all existing tails
 	existingTails := make(map[string]struct{}, len(et))
-	for file := range et {
+	removeTails := make([]string, len(et))
+	for file, tailer := range et {
+		fileStat, err := os.Stat(file)
+		if err != nil {
+			//level.Error(t.logger).Log("msg", "failed to get file stats", "file", file.path, "error", err)
```
Comment (sandeepsukhani) [inline pkg/promtail/targets/filetarget.go]: This should not be commented out, right?

File: pkg/promtail/targets/filetarget.go [hunk 2]
```diff
@@ -304,7 +306,16 @@ func (t *FileTarget) stopTailing(ps []string) {
...
+		}
+		pos, _ := tailer.positions.Get(file)
+		if time.Now().Sub(fileStat.ModTime()) > 10*time.Minute && (pos == fileStat.Size()) {
```
Comment (sandeepsukhani) [inline pkg/promtail/targets/filetarget.go]: I don't see config getting used here. The waiting period is hardcoded instead.
Can you please fix it?

Comment (adityacs) [reply pkg/promtail/targets/filetarget.go]: Hey @sandeepsukhani. Thanks for the review.
Yeah, this is hard coded. This was just a test. Will clean this up

**Comment by adityacs** [comment]
@sandeepsukhani Agree. May be we should close this PR? 

As @slim-bean mentioned here #1761, if there is file rolling, this is automatically taken care of. 

I don't think we would need this feature unless there is a strong reason why someone doesn't want do file rolling.

**Comment by slim-bean** [comment]
Sorry for the slow response on this, it's been difficult for me to make up my mind here.  It's easy to say to people they should change how their environment works but I realize that's not always easy to do.  

That being said, there are quite a few edge cases with supporting this and it would be difficult to get it right, and it's difficult for me to choose adding more complexity to the project when existing solutions for rotating logs are battle tested and better suited to solve this problem.

I'm going to close this PR and the corresponding issue with the guidance that people really should lean on using log rotating solutions and/or schemes that allow promtail to work as built today.

**Pull request closed without merging**